libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -922,6 +1024,64 @@ string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::op
 	});
 }
 
+string YulUtilFunctions::writeToMemoryFunction(Type const& _type)
+{
+	string const functionName =
+		string("write_to_memory_") +
+		_type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&] {
+		solAssert(!dynamic_cast<StringLiteralType const*>(&_type), "");
+		if (auto ref = dynamic_cast<ReferenceType const*>(&_type))
+		{
+			solAssert(
+				ref->location() == DataLocation::Memory,
+				"Can only update types with location memory."
+			);
+
+			return Whiskers(R"(
+				function <functionName>(memPtr, value) {
+					mstore(memPtr, value)
+			}
+			)")
+			("functionName", functionName)
+			.render();
+		}
+		else if (
+			_type.category() == Type::Category::Function &&
+			dynamic_cast<FunctionType const&>(_type).kind() == FunctionType::Kind::External
+		)
+		{
+			return Whiskers(R"(
+				function <functionName>(memPtr, addr, selector) {
+					mstore(memPtr, <combine>(addr, selector))
+				}
+			)")
+			("functionName", functionName)
+			("combine", combineExternalFunctionIdFunction())
```

Comment:
Does this perform cleanup?

Reply:
No, but I think we said we leave that for now as there was no function for cleaning up addr/selector. The evm code was only cleaning up addr I think and it used `FunctionType` for it..

Reply:
Ah, the function actually already performs cleanup.